plutus-scb/src/Plutus/SCB/Core.hs

```diff
@@ -62,78 +62,103 @@ import qualified Data.Set                                   as Set
                                                              Source (..), activeContract, activeContractId,
                                                              activeContractPath, contractPath, dbConfigFile,
                                                              dbConfigPoolSize, hooks, newState,
                                                              partiallyDecodedResponse, toUUID)
-import           Plutus.SCB.Utils                           (liftError, render, tshow)
-import           Wallet.API                                 (ChainIndexAPI, NodeAPI, SigningProcessAPI, WalletAPI,
-                                                             WalletDiagnostics)
+import           Plutus.SCB.Utils                           (render, tshow)
 import qualified Wallet.API                                 as WAPI
-
-newtype Connection =
-    Connection (SqlEventStoreConfig SqlEvent JSONString, ConnectionPool)
+import           Wallet.Emulator.MultiAgent                 (EmulatedWalletEffects)
+
+import           Cardano.Node.Follower                      (NodeFollowerEffect, getBlocks, newFollower)
+import           Plutus.SCB.Effects.Contract                (ContractCommand (..), invokeContract)
+import           Plutus.SCB.Effects.EventLog                (Connection (..), addProcessBus, refreshProjection,
+                                                             runCommand, runGlobalQuery)
+import qualified Plutus.SCB.Effects.EventLog                as EventLog
+import           Plutus.SCB.Effects.UUID                    (uuidNextRandom)
+import           Wallet.API                                 (WalletEffect)
+import           Wallet.Effects                             (ChainIndexEffect, NodeClientEffect, SigningProcessEffect)
+import qualified Wallet.Effects                             as WalletEffects
+
+type ContractEffects =
+        '[ EventLogEffect ChainEvent
+         , Log
+         , UUIDEffect
+         , ContractEffect
+         , Error SCBError
+         ]
 
 installContract ::
-       (MonadLogger m, MonadEventStore ChainEvent m) => FilePath -> m ()
+    ( Member Log effs
+    , Member (EventLogEffect ChainEvent) effs
+    )
+    => FilePath
+    -> Eff effs ()
 installContract filePath = do
-    logInfoN $ "Installing: " <> tshow filePath
+    logInfo $ "Installing: " <> tshow filePath
     void $
         runCommand
             installCommand
             UserEventSource
             (Contract {contractPath = filePath})
-    logInfoN "Installed."
+    logInfo "Installed."
+
+lookupContract ::
+    ( Member (EventLogEffect ChainEvent) effs
+    , Member (Error SCBError) effs
+    )
+    => FilePath
+    -> Eff effs Contract
+lookupContract filePath = do
+    installed <- installedContracts
+    let matchingContracts =
+            Set.filter
+                (\Contract {contractPath} -> contractPath == filePath)
```

Comment:
I think we should give them some other name. Either require the user to supply one or generate one and return it when it's installed.

Reply:
yeah true although that's for another pull request